lib/execution_engine2/sdk/EE2Status.py

```diff
@@ -539,6 +548,27 @@ def _send_exec_stats_to_catalog(self, job_id):
 
         self.sdkmr.catalog_utils.catalog.log_exec_stats(log_exec_stats_params)
 
+    def abandon_children(self, parent_job_id, child_job_ids, as_admin=False) -> Dict:
```

Comment:
Can `get_job_with_permission` be run on a child job ID? Is it necessary to have the parent job ID submitted as part of this request?

I would think that, to cover all bases efficiently, we'd want:
- abandon_job(job_id): given a job_id, kill it -- presumably we have this one already
- abandon_jobs(job_ids): given a set of job_ids, abandon them
- abandon_all_children(parent_job_id): given a parent ID, kill all the children -- requires ability to find parent from a child ID

Is that the plan?

Reply:
@ialarmedalien So this endpoint is more for the "Retry" functionality, it doesn't cancel any jobs. It's meant for abandoning a list of jobs from a batch job on retry. It may not end up being a publicly accessible api.  (I wonder if this actually should spawn a kafka event actually)

As for
*abandon_job and abandon_jobs, we should just use the cancel_jobs() endpoint to terminate all of the child jobs, since we have a list of them already by checking the status of the parent job. Their children will automatically be terminated as well.
abandon_all_children is the same as calling cancel_job on the parent job

Reply:
Maybe its best actually to replace the children instead of abandoning them... And I wonder if we should be submitting based off of the list of dictionaries in the cell, or the saved_job_inputs (which have frozen a particular catalog version already)